post_processing/compare_outputs.jl

```diff
@@ -0,0 +1,91 @@
+import ArgParse
+using Test
+import ClimaAtmos as CA
+import ClimaComms
+import ClimaCore.InputOutput as InputOutput
+import ClimaCore.Fields as Fields
+function parse_commandline()
+    s = ArgParse.ArgParseSettings()
+    ArgParse.@add_arg_table s begin
+        "--output_folder_1"
+        help = "Output file for job 1"
+        arg_type = String
+        "--output_folder_2"
+        help = "Output file for job 2"
+        arg_type = String
+        "--t_end"
+        help = "Simulation end time. This must match the target jobs"
+        arg_type = String
+        "--state_working"
+        help = "A bool indicating that the state test is working [`true` [default], `false`]"
+        arg_type = Bool
+        default = true
+        "--diagnostics_working"
+        help = "A bool indicating that the diagnostics test is working [`true` [default], `false`]"
+        arg_type = Bool
+        default = true
+    end
+    parsed_args = ArgParse.parse_args(ARGS, s)
+    return (s, parsed_args)
+end
+
+(s, parsed_args) = parse_commandline()
+t_end = CA.time_to_seconds(parsed_args["t_end"])
+day = floor(Int, t_end / (60 * 60 * 24))
+sec = floor(Int, t_end % (60 * 60 * 24))
+
+function get_data(folder, name)
+    reader = InputOutput.HDF5Reader(
+        joinpath(folder, "day$day.$sec.hdf5"),
+        ClimaComms.SingletonCommsContext(ClimaComms.CPUSingleThreaded()),
+    )
+    return InputOutput.read_field(reader, name)
+end
+Y_1 = get_data(parsed_args["output_folder_1"], "Y");
+Y_2 = get_data(parsed_args["output_folder_2"], "Y");
+D_1 = get_data(parsed_args["output_folder_1"], "diagnostics");
+D_2 = get_data(parsed_args["output_folder_2"], "diagnostics");
+
+compare(a::FV, b::FV, pn0 = "") where {FV <: Fields.FieldVector} =
+    compare(true, a, b, pn0)
+function _compare(zero_error, a::FV, b::FV, pn0 = "") where {FV}
+    for pn in propertynames(a)
```

Comment:
Should we assert first that the lengths of the `propertynames` for both `a` and `b`, or the lengths of the FieldVectors are the same?

Reply:
They are enforced by type diagonalization